protocol/activity_center_persistence.go

```diff
@@ -0,0 +1,273 @@
+package protocol
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/common"
+)
+
+func (db sqlitePersistence) DeleteActivityCenterNotification(id []byte) error {
+
+	_, err := db.db.Exec(`DELETE FROM activity_center_notifications WHERE id = ?`, id)
+	return err
+}
+
+func (db sqlitePersistence) SaveActivityCenterNotification(notification *ActivityCenterNotification) error {
+	var tx *sql.Tx
+	var err error
+
+	err = notification.Valid()
+	if err != nil {
+		return err
+	}
+
+	tx, err = db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
+	}()
+
+	if notification.Type == ActivityCenterNotificationTypeNewOneToOne {
+		// Delete other notifications so it pop us again if not currently dismissed
+		_, err = tx.Exec(`DELETE FROM activity_center_notifications WHERE id = ? AND dismissed`, notification.ID)
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = tx.Exec(`INSERT INTO activity_center_notifications (id,timestamp,notification_type, chat_id) VALUES (?,?,?,?)`, notification.ID, notification.Timestamp, notification.Type, notification.ChatID)
+	return err
+}
+
+func (db sqlitePersistence) unmarshalActivityCenterNotificationRows(rows *sql.Rows) (string, []*ActivityCenterNotification, error) {
+	var notifications []*ActivityCenterNotification
+	latestCursor := ""
+	for rows.Next() {
+		var chatID sql.NullString
+		var lastMessageBytes []byte
+		var name sql.NullString
+		notification := &ActivityCenterNotification{}
+		err := rows.Scan(
+			&notification.ID,
+			&notification.Timestamp,
+			&notification.Type,
+			&chatID,
+			&notification.Read,
+			&notification.Accepted,
+			&notification.Dismissed,
+			&lastMessageBytes,
+			&name,
+			&latestCursor)
+		if err != nil {
+			return "", nil, err
+		}
+		if chatID.Valid {
+			notification.ChatID = chatID.String
+
+		}
+
+		if name.Valid {
+			notification.Name = name.String
+		}
+
+		// Restore last message
+		if lastMessageBytes != nil {
+			message := &common.Message{}
+			if err = json.Unmarshal(lastMessageBytes, message); err != nil {
+				return "", nil, err
+			}
+			notification.LastMessage = message
+		}
+
+		notifications = append(notifications, notification)
+	}
+
+	return latestCursor, notifications, nil
+
+}
+func (db sqlitePersistence) buildActivityCenterQuery(tx *sql.Tx, cursor string, limit int, ids []types.HexBytes) (string, []*ActivityCenterNotification, error) {
+	args := []interface{}{}
```

Comment:
A little thing, generally an empty slice should be declared via a variable rather than via a literal. It is a small optimisation as you don't cause memory allocation until the slice is written to.

```suggestion
	var args []interface{}
```